File: IS_Lab2_JSON/Python/deserialize_json.py
# -*- coding: utf-8 -*-
"""
deserialize json
"""
import json
class DeserializeJson:
    # konstruktor
    def __init__(self, filename):
        print("let's deserialize something")
        tempdata = open(filename, encoding="utf8")
        self.data = json.load(tempdata)
    # przykładowe statystyki
    def somestats(self):
        example_stat = 0
        for dep in self.data:
            if dep['typ_JST'] == 'GM' and dep['Województwo'] == 'dolnośląskie':
                example_stat += 1
        print('liczba urzędów miejskich w województwie dolnośląskim: ' + ' ' + str(example_stat))
    def allstats(self):
        statystyki = {}
        for dep in self.data:
            wojewodztwo = dep['Województwo'].strip()
            urzad = dep['typ_JST']
            if wojewodztwo not in statystyki:
                statystyki[wojewodztwo] = {}
            if urzad not in statystyki[wojewodztwo]:
                statystyki[wojewodztwo][urzad] = 0
            statystyki[wojewodztwo][urzad] += 1
        for key in statystyki:
            print(f"{key}: {statystyki[key]}")

File: IS_Lab2_JSON/DeserializeJson.cs
using Newtonsoft.Json;


namespace IS_Lab2_JSON
{
    public class DeserializeJson
    {
        public static void Run(string filePath)
        {
            List<AdministrativeUnit> units = JsonConvert.DeserializeObject<List<AdministrativeUnit>>(File.ReadAllText(filePath));
        }
        public static List<AdministrativeUnit> Run1(string filePath)
        {
            return JsonConvert.DeserializeObject<List<AdministrativeUnit>>(File.ReadAllText(filePath));
        }
        public static List<AdministrativeUnit> Deserialize(string filePath)
        {
            return JsonConvert.DeserializeObject<List<AdministrativeUnit>>(File.ReadAllText(filePath));
        }
        public static void AllStats(string filePath)
        {
            Dictionary<string, Dictionary<string, int>> statystki = new Dictionary<string, Dictionary<string, int>>();
            List<AdministrativeUnit> units = JsonConvert.DeserializeObject<List<AdministrativeUnit>>(File.ReadAllText(filePath));
            foreach(var unit in units)
            {
                string wojewodztwo = unit.Województwo.Trim();
                if (!statystki.ContainsKey(wojewodztwo))
                {
                    statystki[wojewodztwo] = new Dictionary<string, int>();
                }
                if (!statystki[wojewodztwo].ContainsKey(unit.typ_JST))
                {
                    statystki[wojewodztwo][unit.typ_JST] = 0;
                }
                statystki[wojewodztwo][unit.typ_JST]++;
            }
            foreach (var wojewodztwo in statystki)
            {
                Console.WriteLine(wojewodztwo.Key);
                foreach (var typ in wojewodztwo.Value)
                {
                    Console.Write($"{typ.Key}: {typ.Value} ");
                }
                Console.WriteLine();
            }
        }
        public static void SomeStats(string filePath)
        {
            int example_stat = 0;
            List<AdministrativeUnit> units = JsonConvert.DeserializeObject<List<AdministrativeUnit>>(File.ReadAllText(filePath));
            foreach (var unit in units)
            {
                if (unit.typ_JST == "GM" && unit.Województwo == "dolnośląskie")
                {
                    example_stat++;
                }
            }
            Console.WriteLine("liczba urzędów miejskich w województwie dolnośląskim: " + " " + example_stat);
        }
    }
}